src/dataclay/data_manager.py

```diff
@@ -143,7 +143,7 @@ async def load_object(self, instance: DataClayObject):
                 )
                 self.dataclay_stored_objects.dec()
             except Exception as e:
-                raise DataClayException("Object not found.") from e
+                raise ObjectError("Object not found.") from e
```

Comment:
This instantiation of ObjectError confuses as the `__init__` says that parameter is "ObjectID". Maybe use a subclass (e.g. ObjectNotFound) and pass the ID and add a valid `__str__` method in the ObjectNotFound class? (Class that will inherit from ObjectError).